x/ibc/04-channel/types/msgs.go

```diff
@@ -15,10 +17,11 @@ import (
 var _ sdk.Msg = MsgChannelOpenInit{}
 
 type MsgChannelOpenInit struct {
-	PortID    string         `json:"port_id"`
-	ChannelID string         `json:"channel_id"`
-	Channel   Channel        `json:"channel"`
-	Signer    sdk.AccAddress `json:"signer"`
+	PortID    string                `json:"port_id"`
+	ChannelID string                `json:"channel_id"`
+	PortCap   capability.Capability `json:"port_capability"`
```

Comment:
Putting capabilities in the user msgs here to make everything build, though I don't think it's possible for an end user outside of the state machine to ever pass in a capability correctly.

Either this functionality needs to be removed or it needs to somehow be routed first through the appropriate module that can inject the capability on behalf of the user

Reply:
It needs to be routed through the module, see above comment.